server/src/main/java/org/opensearch/action/bulk/BulkItemResponse.java

```diff
@@ -96,6 +96,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field(_ID, failure.getId());
             builder.field(STATUS, failure.getStatus().getStatus());
             builder.startObject(ERROR);
+            builder.field(_INDEX, failure.getIndex());
```

Comment:
Adding these new fields will add to the b/w usage for existing users who are not using filter path. I would suggest that we rather include this in the error reason in a way that this is backward compatible. You can also consider adding a new field that can be controlled by query parameter similar to what we have in `_cat/nodes` api where we can control which fields are returned.

Reply:
By default we should not be increasing the response size and it should be controlled by the user that they need the additional information that you have added here.